apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -58,6 +60,34 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
   const sdlBuilderRef = useRef<SdlBuilderRefType>(null);
   const { minDeposit } = useChainParam();
 
+  const nextRouter = useNextRouter();
+  const { templateId } = nextRouter.query;
+
+  const fileUploadRef = useRef<HTMLInputElement>(null);
+
+  const handleFileChange = event => {
+    const fileUploaded = event.target.files[0];
+    const reader = new FileReader();
+
+    reader.onload = event => {
+      setSelectedTemplate({
+        title: "From file",
+        code: "from-file",
+        category: "General",
+        description: "Custom uploaded file",
+        content: event.target?.result as string
+      });
+      setEditedManifest(event.target?.result as string);
+      router.push(UrlService.newDeployment({ step: RouteStepKeys.editDeployment }));
+    };
+
+    reader.readAsText(fileUploaded);
+  };
+
+  async function fromFile() {
```

Comment:
I'm not sure I understand what this name means. Would you like this one?

```suggestion
  async function triggerFileInput() {
```